### PR TITLE
fix: Inspector method overrides when contextIsolation enabled

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Closes https://github.com/electron/electron/issues/29543.

Introduced with the default change from `contextIsolation: false` to `contextIsolation: true`.  When `contextIsolation` is enabled, `window.InspectorFrontendHost`, `window.Persistence`, and `window.UI` are all undefined in the isolated world. We therefore need to set them with the `contextBridge` so that we can override them correctly.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Fixed an issue where dropdown menus did not work in DevTools when `contextIsolation` was enabled.

### CODE REVIEW

**Review by deepak1556 — APPROVED** [review]

**Review by MarshallOfSound — APPROVED** [review]

File: lib/renderer/inspector.ts [hunk 1]
```diff
@@ -36,9 +52,18 @@ const createMenu = function (x: number, y: number, items: ContextMenuItem[]) {
   const isEditMenu = useEditMenuItems(x, y, items);
   ipcRendererInternal.invoke<number>(IPC_MESSAGES.INSPECTOR_CONTEXT_MENU, items, isEditMenu).then(id => {
     if (typeof id === 'number') {
-      window.DevToolsAPI!.contextMenuItemSelected(id);
+      if (contextIsolationEnabled) {
+        webFrame.executeJavaScript(`window.DevToolsAPI.contextMenuItemSelected(${JSON.stringify(id)})`);
+      } else {
+        window.DevToolsAPI!.contextMenuItemSelected(id);
+      }
```
Comment (nornagon) [inline lib/renderer/inspector.ts:59]: I think the `webFrame.executeJavaScript` version works fine regardless of the value of `contextIsolationEnabled`, right?

Possibly `internalContextBridge.overrideGlobalValueFromIsolatedWorld` works fine in non-contextIsolation situations too, and we can drop all these conditionals.

Comment (codebytere) [reply lib/renderer/inspector.ts:59]: @nornagon i'm happy with either approach, though i took this one given security hazards of `webFrame.executeJavaScript` and the fact that we're needing to do string templating in there

Comment (nornagon) [reply lib/renderer/inspector.ts:59]: If there are security hazards, we shouldn't be using it at all, regardless of whether the context is isolated :) What you have there seems OK to me, what sort of hazard are you thinking of?

Comment (codebytere) [reply lib/renderer/inspector.ts:59]: just the general risk of executing js with templated strings  - you're right though, it's fine enough and i'll make the change!

**Review by nornagon — APPROVED** [review]
would also be good to check if we can remove the conditional in `window.onload` as well, to reduce the number of codepaths.

**Comment by codebytere** [comment]
Failure not germane - merging.

**Pull request merged**